src/drivers/intel/hda/hda-dma.c

```diff
@@ -322,10 +322,10 @@ static inline int hda_dma_is_buffer_empty(struct dma_chan_data *chan)
 
 static int hda_dma_wait_for_buffer_full(struct dma_chan_data *chan)
 {
-	uint64_t deadline = k_cycle_get_64() + k_us_to_cyc_ceil64(HDA_DMA_TIMEOUT);
+	uint64_t deadline = sof_cycle_get_64() + k_us_to_cyc_ceil64(HDA_DMA_TIMEOUT);
 
 	while (!hda_dma_is_buffer_full(chan)) {
-		if (deadline < k_cycle_get_64()) {
+		if (deadline < sof_cycle_get_64()) {
```

Comment:
So this is one category of usage. This seems not to be about precision, but there's inbuilt assumption that k_cycle_get_64 will not roll over. Even with a 1Ghz clock, this would still take 500+ years, so I guess this is reasonable. Personally I'd prefer an explicit rollover check still.

Reply:
@kv2019i should i add your comments in the code to explain why are we using 64 bit counters? Can do that in a separate patch.

Please note that the code stays the same for platforms that do have a hardware register counting 64bits cycles.

Reply:
Probably no need as we are not changing the semantics in the PR.